gcm_filters/kernels.py

```diff
@@ -531,6 +520,25 @@ def __post_init__(self):
         # note: wet_mask(j+1,i) corresponds to np.roll(wet_mask, -1, axis=-2)
         self.n_wet_mask = self.wet_mask * np.roll(self.wet_mask, -1, axis=-2)
 
+        # check that northern edge grid data folds onto itself if not on land;
+        # note: grid data goes crazy for POP model land points so we don't want to check for land points
+        nx = np.shape(self.dxn)[-1]  # number of longitudes or columns
+        # grab second to last row since we have already appended one extra row
+        first_half = (self.n_wet_mask * self.dxn)[..., [-2], : (nx // 2)]
```

Comment:
It might be better to do this test with boolean indexing rather than multiplying by the wet mask. If, for example, some of the land data is `NaN` then I think this test would throw an error because `0 * np.NaN` is `np.NaN` and `np.NaN == np.NaN` returns `False`. Maybe something like
`first_half = self.dxn[...,-2,self.n_wet_mask[...,-2,:(nx//2)] == 1]`
would avoid NaN problems.

Reply:
Thanks for the catch! In my latest commit I followed your suggestion - just using a different syntax that will also work with dask and cupy.